root.c

```diff
@@ -498,12 +498,12 @@ bool w_root_process_pending(w_root_t *root,
   }
 
   w_log(W_LOG_DBG, "processing %d events in %s\n",
-      w_ht_size(coll->pending_uniq), root->root_path->buf);
+      w_pending_coll_size(coll), root->root_path->buf);
 
   // Steal the contents
   pending = coll->pending;
   coll->pending = NULL;
-  w_ht_free_entries(coll->pending_uniq);
+  w_pending_coll_drain(coll);
```

Comment:
This is fixing a mild layering violation.  This bit of code wants to steal the linked list of pending items, but wants to make sure that the pending list doesn't reference them any more.  Previously this was done by ensuring that the pending_uniq hash got emptied, but since that hash was removed by this diff it can't do that any more.

Really this code should have been calling `w_pending_coll_drain`; that function pops anything from the list (which will be empty in this case, because we just stole it) and then empties the hash.

After this diff, we use the `art` instead of the hash.  There isn't an equivalent function to `w_ht_free_entries` for `art`, instead we have to destroy and re-initialize; we could chose to continue the layering violation here, but it would look uglier :-)